geth/jail/cell.go

```diff
@@ -110,4 +110,30 @@ func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) error {
 	case <-timer.C:
 		return errors.New("Timeout")
 	}
+
+}
+
+// Set calls Set on the underlying JavaScript VM.
+func (c *Cell) Set(key string, val interface{}) error {
```

Comment:
Is this method called somewhere outside the `jail` package? If so, it needs to be thread-safe. If not — it shouldn't be in the interface. The same is true for every other method here.

E.g. returning `otto.Value` isn't thread-safe because it contains a vm with no mutexes inside.

Reply:
Ohhhhh, I think  I see now. Thanks for the example. I will double check every method on the interface and document what I find here.

Reply:
From what I can see, status-go does not directly consume the `JailCell` interface? (`jail.Jail` is responsible for creating cells via `CreateCell` and `Cell` to retrieve cells, but neither seem to be directly called in status-go) - Are they invoked via RPC? Pardon my unfamiliarity with the architecture :)

Reply:
@alexkohler I don't see a lot of usages of this interface, I think it primarily is used for mocking. Theoretically, we can nuke it and use something else in tests, but it would not be related to this issue.

Reply:
@mandrigin wasn't part of your original comment to tighten the `JailCell` interface if these methods weren't called outside the `jail` package? And yes, I'll double check on the thread safety of the methods on `Cell`. Thank you for the prompt replies!

Reply:
Yeah, maybe I was misguided by this interface as well :)

Reply:
From what I can see, Cells are created via `NewCell` in `geth/jail/jail.go`. This is called only by `geth/jail`'s `obtainCell`. `CreateCell` is the only caller of `obtainCell`, which returns a `JailCell` (which we already concluded it doesn't appear to be used in the actual implementation). 

This being said, it doesn't look like `Cell` nor `JailCell`'s methods are called outside of the jail package. Would you prefer making them threadsafe (in case they are ever used) or just removing them from `Cell` and tightening the interface?

Reply:
I'd both tighten them and wrap `otto.Value` with some wrapper that has a method that returns underlying value but at least has a comment that it is unsafe.

Reply:
@mandrigin I think we can safely get rid of `Get` and `Set` which then leaves us with the following interface:
```Go
type JailCell interface {
	// Run an arbitrary JS code. Input maybe string or otto.Script.
	Run(interface{}) (VMValue, error)
	// Call an arbitrary JS function by name and args.
	Call(item string, this interface{}, args ...interface{}) (VMValue, error)
	// Stop stops background execution of cell.
	Stop() error
}
```
However, at this point, it appears we're straying away from what `JailCell` was originally meant for. I'm thinking it might make sense to leave `Get` and `Set` around and simply provide a wrapper around any `otto.Value`s returned - thoughts?

Reply:
@alexkohler sounds sensible to me 👍

Reply:
So I'm running into a little trouble - if we declare the wrapper inside of `geth/jail`, this means the `JailCell` interface in `geth/common/types.go` will now cause us to import `geth/jail`, which causes an import cycle.

If we declare the wrapper inside of `geth/common/types.go`, we'd have to export the field the struct was wrapping (I was hoping to just export the wrapper) so `Cell` could use it down in `geth/jail`. This also doesn't seem great. Any ideas?

Reply:
is it possible to move `JailCell` to `geth/jail`? is it a lot of work?

Reply:
@mandrigin With a cursory glance, it doesn't look like a lot of work - we'd have to also move the `JailManager` interface, and the `MockJailCell`/`MockJailManager` down into `geth/jail` as well. I can give that a try.